### PR TITLE
glib-openssl 2.50.4 (new formula)

### PR DESCRIPTION
**Pull request by tschoonj**

### CODE REVIEW

**Comment by ilovezfs** [comment]
@BrewTestBot test this please

**Comment by tschoonj** [comment]
@ilovezfs can I merge this one in?

File: Formula/glib-openssl.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+          return 0;
+        else
+          return 1;
+      }
+    EOS
+
+    # From `pkg-config --cflags --libs gio-2.0`
+    flags = [
+      "-D_REENTRANT",
+      "-I#{HOMEBREW_PREFIX}/include/glib-2.0",
```
Comment (ilovezfs) [inline Formula/glib-openssl.rb]: why not use the Formula opt_include?

File: Formula/glib-openssl.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  desc "OpenSSL GIO module for glib"
+  homepage "https://launchpad.net/glib-networking"
+  url "https://download.gnome.org/sources/glib-openssl/2.50/glib-openssl-2.50.4.tar.xz"
+  sha256 "eda914e29d9d72f03185aba92c33ec3949862c05b73b8eb04d1e1b6217b756c0"
+
+  depends_on "pkg-config" => :build
+  depends_on "glib"
+  depends_on "openssl"
+
+  link_overwrite "lib/gio/modules"
```
Comment (ilovezfs) [inline Formula/glib-openssl.rb]: why is this needed

File: Formula/glib-openssl.rb [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+  def install
+    # Install files to `lib` instead of `HOMEBREW_PREFIX/lib`.
+    inreplace "configure", "$($PKG_CONFIG --variable giomoduledir gio-2.0)", lib/"gio/modules"
```
Comment (ilovezfs) [inline Formula/glib-openssl.rb]: there's no configure arg or envvar for this?

**Comment by tschoonj** [comment]
Honestly this formula is mostly copy-paste from `glib-networking`. I don't even know what the `link_overwrite` does but I guess it has to do with the `post_install` phase which will create a file in this folder.

File: Formula/glib-openssl.rb (hunk 3 above)
Comment (tschoonj) [reply Formula/glib-openssl.rb]: nope

File: Formula/glib-openssl.rb (hunk 1 above)
Comment (tschoonj) [reply Formula/glib-openssl.rb]: Let me fix that...

File: Formula/glib-openssl.rb (hunk 3 above)
Comment (ilovezfs) [reply Formula/glib-openssl.rb]: ok

File: Formula/glib-openssl.rb (hunk 1 above)
Comment (ilovezfs) [reply Formula/glib-openssl.rb]: ditto with the -L stuff below

**Review by ilovezfs — APPROVED** [review]

**Comment by ilovezfs** [comment]
looks good modulo the nits above

**Comment by DomT4** [comment]
> I don't even know what the link_overwrite does but I guess it has to do with the post_install phase which will create a file in this folder.

It shouldn't be needed for new formula, by this point. Documentation is [here](https://github.com/Homebrew/brew/blob/76cd7c79498daf380d99ed7379394618164b602b/Library/Homebrew/formula.rb#L1031-L1065). It came about as of the discussion [here](https://github.com/Homebrew/legacy-homebrew/issues/41501#issuecomment-130959611). Theoretically at this point you could probably remove it from some/all existing formulae, making the argument people will have likely updated their installations, but 🤷‍♂️.

**Comment by DomT4** [comment]
The other use case is [this sort of thing](https://github.com/Homebrew/homebrew-core/commit/949ae5c935b), but again, you could make the argument people will have likely run `brew upgrade` or `brew upgrade mono` at some point in the last two years & the workaround is safe to remove. It wasn't really _supposed_ to be a permanent addition to formulae, IMO.

**Comment by tschoonj** [comment]
I removed the `link_overwrite` line. Thanks for the info!

**Comment by ilovezfs** [comment]
:icecream:
